pkg/auth/handler/webapp/settings.go

```diff
@@ -68,91 +126,77 @@ func (h *SettingsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	defer ctrl.Serve()
 
 	redirectURI := httputil.HostRelative(r.URL).String()
-	authenticatorID := r.Form.Get("x_authenticator_id")
+	providerAlias := r.Form.Get("x_provider_alias")
+	identityID := r.Form.Get("x_identity_id")
 	userID := ctrl.RequireUserID()
+	nonceSource, _ := r.Cookie(h.CSRFCookie.Name)
 
 	ctrl.Get(func() error {
-		data := map[string]interface{}{}
-
-		baseViewModel := h.BaseViewModel.ViewModel(r, nil)
-		viewmodels.Embed(data, baseViewModel)
-
-		authenticators, err := h.Authenticators.List(userID)
+		data, err := h.GetData(r, w)
 		if err != nil {
-			panic(err)
-		}
-
-		totp := false
-		oobotp := false
-		password := false
-		for _, typ := range h.Authentication.SecondaryAuthenticators {
-			switch typ {
-			case authn.AuthenticatorTypePassword:
-				password = true
-			case authn.AuthenticatorTypeTOTP:
-				totp = true
-			case authn.AuthenticatorTypeOOB:
-				oobotp = true
-			}
-		}
-
-		viewModel := SettingsViewModel{
-			Authenticators:           authenticators,
-			SecondaryTOTPAllowed:     totp,
-			SecondaryOOBOTPAllowed:   oobotp,
-			SecondaryPasswordAllowed: password,
+			return err
 		}
-		viewmodels.Embed(data, viewModel)
 
 		h.Renderer.RenderHTML(w, r, TemplateWebSettingsHTML, data)
 		return nil
 	})
 
-	ctrl.PostAction("revoke_devices", func() error {
-		if err := h.MFA.InvalidateAllDeviceTokens(userID); err != nil {
+	ctrl.PostAction("link_oauth", func() error {
```

Comment:
Seems we don't need `link_oauth` action in `/settings`, since we won't show the oauth identity candidates